packages/hooks/src/useWalletMigration.ts

```diff
@@ -0,0 +1,29 @@
+import { useCallback } from 'react';
+import { useFetch, useInvalidateQuery, useRequest } from '@deriv/api';
+
+const useWalletMigration = () => {
+    const invalidate = useInvalidateQuery();
+
+    /** Make a request to wallet_migration API and onSuccess it will invalidate the cached data  */
+    const { mutate } = useRequest('wallet_migration', { onSuccess: () => invalidate('wallet_migration') });
+
+    /** Fetch the wallet_migration API and refetch it every second if the status is in_progress */
+    const { data } = useFetch('wallet_migration', {
+        payload: { wallet_migration: 'status' },
+        options: {
+            refetchInterval: response => (response?.wallet_migration?.status === 'in_progress' ? 1000 : false),
+        },
+    });
+
+    const start = useCallback(() => mutate({ payload: { wallet_migration: 'start' } }), [mutate]);
+
+    const reset = useCallback(() => mutate({ payload: { wallet_migration: 'reset' } }), [mutate]);
+
+    return {
+        status: data?.wallet_migration?.status,
+        start,
+        reset,
+    };
```

Comment:
```suggestion
    const status = data?.wallet_migration?.status;
    
    return {
        /** The status of the wallet migration */
        status,
        /** Whether the wallet migration is eligible */
        is_eligible: status === 'eligible',
        /** Whether the wallet migration is failed */
        is_failed: status === 'failed',
        /** Whether the wallet migration is in progress */
        is_in_progress: status === 'in_progress',
        /** Whether the wallet migration is ineligible */
        is_ineligible: status === 'ineligible',
        /** Whether the wallet migration is completed */
        is_migrated: status === 'migrated',
        /** Sends a request to the server to start the wallet migration process */
        start,
        /** Sends a request to the server to reset the wallet migration process */
        reset,
    };
```
@thisyahlen-deriv Improve the comments please 🙏🏻👀